service/history/api/get_workflow_util.go

```diff
@@ -234,6 +248,42 @@ func GetMutableState(
 	return MutableStateToGetResponse(mutableState)
 }
 
+func GetMutableStateWithConsistencyCheck(
+	ctx context.Context,
+	shardContext shard.Context,
+	workflowKey definition.WorkflowKey,
+	currentVersion int64,
+	currentEventID int64,
+	workflowConsistencyChecker WorkflowConsistencyChecker,
+) (_ *historyservice.GetMutableStateResponse, retError error) {
+
+	if len(workflowKey.RunID) == 0 {
+		return nil, serviceerror.NewInternal(fmt.Sprintf(
+			"getMutableState encountered empty run ID: %v", workflowKey,
+		))
+	}
+
+	workflowLease, err := workflowConsistencyChecker.GetWorkflowLeaseWithConsistencyCheck(
+		ctx,
+		nil,
+		func(mutableState workflow.MutableState) bool {
+			return currentVersion < mutableState.GetCurrentVersion() && currentEventID < mutableState.GetNextEventID()
```

Comment:
version can be equal right?

Reply:
mutable state current version may not be the last event version? Shall we use last item of the current version history instead?

Reply:
updated